src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Vector256.cs

```diff
@@ -171,6 +173,35 @@ public static Vector256<ulong> AsUInt64<T>(this Vector256<T> vector)
             return vector.As<T, ulong>();
         }
 
+        /// <summary>Reinterprets a <see cref="Vector{T}" /> as a new <see cref="Vector256{T}" />.</summary>
+        /// <typeparam name="T">The type of the vectors.</typeparam>
+        /// <param name="value">The vector to reinterpret.</param>
+        /// <returns><paramref name="value" /> reinterpreted as a new <see cref="Vector256{T}" />.</returns>
+        /// <exception cref="NotSupportedException">The type of <paramref name="value" /> (<typeparamref name="T" />) is not supported.</exception>
+        public static Vector256<T> AsVector256<T>(this Vector<T> value)
+            where T : struct
+        {
+            Debug.Assert(Vector256<T>.Count >= Vector<T>.Count);
```

Comment:
This currently assumes `Vector<T>` is no bigger than `Vector256<T>`, the logic would need to change if we supported `Vector512` or others...